Main/inc/Analysis.hh

```diff
@@ -101,52 +107,159 @@ namespace roofitter {
+        file->Write();
+
+        return tree_flat;
+    }
 
-      RooArgSet vars;
-      RooRealVar* x_var = 0;
-      RooRealVar* y_var = 0;
-      std::string x_leaf = "";
-      std::string y_leaf = "";
-      for (const auto& i_obs : _observables) {
-	const auto& i_obs_conf = i_obs.getConf();
-	RooRealVar* var = _ws->var(i_obs_conf.name().c_str());
-	vars.add(*var);
-
-	if (x_leaf.empty()) {
-	  x_leaf = i_obs_conf.leaf();
-	  x_var = var;
-	}
-	else if (y_leaf.empty()) {
-	  y_leaf = i_obs_conf.leaf();
-	  y_var = var;
-	}
-      }
+    void fillData(TTree* tree) {
 
-      std::string histname = "h_" + _anaConf.name();
-      std::string draw = "";
-      if (vars.getSize()==1) {
-	_hist = x_var->createHistogram(histname.c_str());
-	draw = x_leaf;
-      }
-      else if (vars.getSize()==2) {
-	_hist = x_var->createHistogram(histname.c_str(), RooFit::YVar(*y_var));
-	draw = y_leaf+":"+x_leaf;
-      }
-      else {
-	throw cet::exception("Analysis::fillData()") << "Can't create histogram with more than two axes";
-      }
-      draw += ">>";
-      draw += _hist->GetName();
+    RooArgSet vars;
+    RooRealVar* x_var = 0;
+    RooRealVar* y_var = 0;
+    std::string x_leaf = "";
+    std::string y_leaf = "";
+    for (const auto& i_obs : _observables) {
+        const auto& i_obs_conf = i_obs.getConf();
+	    RooRealVar* var = _ws->var(i_obs_conf.name().c_str());
+	    vars.add(*var);
+
+	    if (x_leaf.empty()) {
+	        x_leaf = i_obs_conf.leaf();
+	        x_var = var;
+	    }
+	    else if (y_leaf.empty()) {
+	        y_leaf = i_obs_conf.leaf();
+	    y_var = var;
+	    }
+    }
 
-      tree->Draw(draw.c_str(), cutcmd(), "goff");
+    if (_anaConf.fit_type() == "binned")
+    {
+        std::string histname = "h_" + _anaConf.name();
+        std::string draw = "";
+        if (vars.getSize()==1) {
+	        _hist = x_var->createHistogram(histname.c_str());
+	        draw = x_leaf;
+        }
+        else if (vars.getSize()==2) {
+	        _hist = x_var->createHistogram(histname.c_str(), RooFit::YVar(*y_var));
+	        draw = y_leaf+":"+x_leaf;
+        }
+        else {
+            throw cet::exception("Analysis::fillData()") << "Can't create histogram with more than two axes";
+        }
+        draw += ">>";
+        draw += _hist->GetName();
+
+        tree->Draw(draw.c_str(), cutcmd(), "goff");
+
+        _ws->import(*(new RooDataHist("data", "data", vars, RooFit::Import(*_hist))));
+    }
+    else if (_anaConf.fit_type() == "unbinned")
+    {
+        TFile *file_flat = new TFile("trkana_flat.root");
+        TTree *flat_tree = (TTree*) file_flat->Get("trkana_flat");
+        RooRealVar *mom = new RooRealVar("mom", "mom", 95, 115);
 
-      _ws->import(*(new RooDataHist("data", "data", vars, RooFit::Import(*_hist))));
+//        TTreeFormula *cut_formula = new TTreeFormula("cut", cutcmd(), tree);
+        RooDataSet data("data", "unbinned dataset", flat_tree, *mom);
+        _ws->import(data);
     }
```

Comment:
Could you add a final else block that throws an exception? This will stop users from mistyping the fit_type string